test/new-e2e/tests/npm/ec2_1host_test.go

```diff
@@ -0,0 +1,160 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package npm
+
+import (
+	"math"
+	"os"
+	"testing"
+	"time"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+	krpretty "github.com/kr/pretty"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const NPMsystemProbeConfig = `
+network_config:
+  enabled: true
+`
+
+type ec2VMSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+// TestEC2VMSuite will validate running the agent on a single EC2 VM
+func TestEC2VMSuite(t *testing.T) {
+	s := &ec2VMSuite{}
+	e2eParams := []params.Option{}
+	// debug helper
+	if _, devmode := os.LookupEnv("TESTS_E2E_DEVMODE"); devmode {
+		e2eParams = []params.Option{params.WithDevMode(), params.WithSkipDeleteOnFailure()}
+		s.DevMode = true
+	}
+
+	// Source of our kitchen CI images test/kitchen/platforms.json
+	// Other VM image can be used, our kitchen CI images test/kitchen/platforms.json
+	// ec2params.WithImageName("ami-a4dc46db", os.AMD64Arch, ec2os.AmazonLinuxOS) // ubuntu-16-04-4.4
+	e2e.Run(t, s, e2e.FakeIntakeStackDef(nil, agentparams.WithSystemProbeConfig(NPMsystemProbeConfig)), e2eParams...)
+}
+
+// TestFakeIntakeNPM Validate the agent can communicate with the (fake) backend and send connections every 30 seconds
+//   - looking for 1 host to send CollectorConnections payload to the fakeintake
+//   - looking for 3 payloads and check if the last 2 have a span of 30s +/- 500ms
+func (v *ec2VMSuite) TestFakeIntakeNPM() {
+	t := v.T()
+
+	// default is to reset the current state of the fakeintake aggregators
+	if !v.DevMode {
+		v.Env().Fakeintake.FlushServerAndResetAggregators()
+	}
+
+	targetHostnameNetID := ""
+	// looking for 1 host to send CollectorConnections payload to the fakeintake
+	v.EventuallyWithT(func(c *assert.CollectT) {
+		// generate a connection
+		v.Env().VM.Execute("curl http://www.datadoghq.com")
```

Comment:
Do you need to do this in the EventuallyWithT, or is it okay to do once?

Reply:
I'm not sure yet when the agent/system-probe is started, I discuss previously with Paola and we agreed to add a check/fence later